test/compiler/irpasses.jl

```diff
@@ -295,3 +295,20 @@ end
 const _some_coeffs = (1,[2],3,4)
 splat_from_globalref(x) = (x, _some_coeffs...,)
 @test splat_from_globalref(0) == (0, 1, [2], 3, 4)
+
+# External passes
+const _opt = Ref{Any}(nothing)
+avx_pass(opt) = (_opt[] = opt; opt)
+macro avx(ex)
```

Comment:
because this doesn't actually do anything to do with `avx` it would be clearer maybe to call this
`demo_pass` and `@with_demo_pass` ?

Reply:
If we go with the "comment out" option this will be deleted anyway. But yes, for the demo we need better names.